server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Rights.scala

```diff
@@ -140,7 +140,7 @@ case class Rights private(rights: Map[Username, Seq[Right]]) {
   def append(username: Username, rights: Seq[Right]): Rights = {
     require(rights.nonEmpty, "'rights' should not be empty")
```

Comment:
please avoid to Throw, you can return an Either or a Try in worst scenario.

Reply:
Well here I'll relax the assertion as appending empty rights is not harmful imo